net/dns/config.go

```diff
@@ -48,6 +48,12 @@ type Config struct {
 	// it to resolve, you also need to add appropriate routes to
 	// Routes.
 	Hosts map[dnsname.FQDN][]netip.Addr
+	// CnameHosts maps DNS FQDNs to their CNAME targets.
+	// Targets must be domains that can be resolved via Hosts.
+	// Queries matching entries return a CNAME chain followed by the
+	// resolved A/AAAA record.
+	// Wildcard CNAMEs are supported using "*.domain." as the key.
+	CnameHosts map[dnsname.FQDN]dnsname.FQDN
```

Comment:
As I've stated in Slack DMs/calls and that PR, I neither like the name "Cname" for something that's not a CNAME record, nor do I like adding per-node options to the global DNSConfig. That'd mean we'd have to recalculate the whole DNSConfig whenever any one node changes. That doesn't work with large tailnets. (same problem as our packet filters, for example)

So it really needs to be a field on Node. It's logically a boolean ("does this node want *.<[itsname.ts.net](http://itsname.ts.net/)>"?) but don't need to add new fields because we already have a generic extension method for arbitrary key/value pairs on nodes: NodeCap